### PR TITLE
KAFKA-7177: Update 2.0 documentation to reflect changed quota behaviors by KIP-219

### PR DESCRIPTION
**Pull request by jonlee2**
Updated the 2.0 document for changed quota behaviors.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by jonlee2** [comment]
@lindong28 Can you review? Thanks.

File: docs/design.html [hunk 1]
```diff
@@ -610,10 +610,12 @@ <h4><a id="design_quotasenforcement" href="#design_quotasenforcement">Enforcemen
...
-        It computes the amount of delay needed to bring a guilty client under its quota and delays the response for that time. This approach keeps the quota violation transparent to clients
-        (outside of client-side metrics). This also keeps them from having to implement any special backoff and retry behavior which can get tricky. In fact, bad client behavior (retry without backoff)
-        can exacerbate the very problem quotas are trying to solve.
+        How does a broker react when it detects a quota violation? In our solution, the broker first computes the amount of delay needed to bring a client exceeding its quota and
+        returns a response with the delay immediately. In case of a fetch request, the response will not contain any data. Then, the broker mutes the channel to the violating client,
+        not to process requests from the client anymore, until the delay is over. Upon receiving a response with a non-zero delay duration, the Kafka client layer will also refrain from
+        sending further requests to the broker during the delay. This approach keeps the quota violation transparent to clients (outside of client-side metrics). This also keeps them
```
Comment (lindong28) [inline docs/design.html]: We probably should change or remove the sentence that says `This approach keeps the quota violation transparent to clients (outside of client-side metrics). This also keeps them
from having to implement any special backoff and retry behavior which can get tricky. In fact, bad client behavior (retry without backoff) can exacerbate the very problem quotas are trying to solve`.

**Comment by jonlee2** [comment]
Remove those sentences. @lindong28 Can you take a look again? Thanks.

**Comment by lindong28** [comment]
@jonlee2 LGTM in terms of its content. I think `client layer` should probably just be `client`.

@ijuma You have much better english writing than me. Since this touches the design doc that user reads, would you have time to take a look?

**Comment by ijuma** [comment]
Thanks for the PR. I think it would be useful to explain the reasoning of the approach a little. For example, this still handles badly behaved clients by relying on socket buffer back pressure.

**Comment by jonlee2** [comment]
Updated the text as suggested. @lindong28 @ijuma Can you take another look? Thanks.

**Comment by lindong28** [comment]
Thanks for update. LGTM. @ijuma may have better ideas of writing.

**Comment by asfgit** [comment]
FAILURE
 9189 tests run, 5 skipped, 1 failed.
--none--

**Comment by lindong28** [comment]
I just merged it to trunk so that we can use the design doc for 2.0 release. We can have minor followup patch to improve the doc if needed.

**Comment by ijuma** [comment]
Seems fine to me, thanks.

**Pull request closed without merging**